plugins/k8smeta/src/plugin.cpp

```diff
@@ -441,11 +317,53 @@ bool my_plugin::init(falcosecurity::init_input& in)
         return false;
     }
 
-    // Here we do /proc scan to catch the pod_uid from already running
-    // processes.
-    // We cannot populate the sinsp thread table because when we call `init` it
-    // is still empty. The /proc scan in sinsp is done after the plugin init.
-    do_initial_proc_scan();
+    return true;
+}
+
+//////////////////////////
+// Listen capability
+//////////////////////////
+
+bool my_plugin::capture_open(const falcosecurity::capture_listen_input& in) {
+    using st = falcosecurity::state_value_type;
+
+    SPDLOG_DEBUG("enriching initial thread table entries");
+    auto& tr = in.get_table_reader();
+    auto& tw = in.get_table_writer();
+    m_thread_table.iterate_entries(
```

Comment:
We iterate over all thread table entries, and for each, we iterate over its cgroups vector, taking the `->second` (cgroups are a vector of std::pair and `second` hosts the cgroup path, while `first` hosts the cgroup controller); for each cgroup path we try to extract the `pod_uid` and, if successful, we break the loop for the current thread entry.